src/java/org/apache/cassandra/service/accord/AccordVerbHandler.java

```diff
@@ -52,45 +48,34 @@ public void doVerb(Message<T> message) throws IOException
         // TODO (desired): need a non-blocking way to inform CMS of an unknown epoch and add callback to it's receipt
 //        ClusterMetadataService.instance().maybeCatchup(message.epoch());
         logger.debug("Receiving {} from {}", message.payload, message.from());
+        Node.Id fromNodeId = endpointMapper.mappedId(message.from());
         T request = message.payload;
+
         long knownEpoch = request.knownEpoch();
         if (!node.topology().hasEpoch(knownEpoch))
         {
             node.configService().fetchTopologyForEpoch(knownEpoch);
             long waitForEpoch = request.waitForEpoch();
             if (!node.topology().hasEpoch(waitForEpoch))
```

Comment:
This seems like it's a property independent of `knownEpoch` and we could move it out and not have to copy the journal append part and have less indentation? The journal append is not dependent on having `waitForEpoch` it just depends on side effects. It really just changes whether we process directly or `withEpoch`.

Maybe `request.process` should handle this check. instead of the caller?

Reply:
Good point. Simplified. Not touching `request.process()` as part of this though.